packages/contracts/contracts/testing/ERC1155Reward.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@snickerdoodlelabs/erc7529/contract/ERC7529.sol";
+
+contract ERC1155Reward is ERC1155, AccessControl, ERC1155Supply, ERC7529 {
+    bytes32 public constant URI_SETTER_ROLE = keccak256("URI_SETTER_ROLE");
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+
+    // Number of reward options
+    uint256 public currentTokenId; 
+
+    /// @notice Takes the initial number of rewards offered and sets the current token id
+    constructor(uint256 numberOfRewards) ERC1155("") {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _grantRole(MINTER_ROLE, msg.sender);
+
+        // Starts from token id 0
+        currentTokenId = numberOfRewards - 1;
```

Comment:
Is it OK to have the currentTokenId be -1? I assume unit256 auto-initializes to 0. Do we need an assert that numberOfRewards > 0?

Reply:
Also, do we need to do something like take an array of tokenURIs for when we setup the initial rewards?

Reply:
@SnickerChar good points, it does auto-initiate to 0. Will add checks for > 0. 

According to the standard [here](https://eips.ethereum.org/EIPS/eip-1155#metadata-extensions), it seems that uri should always be the same eg. ipfs://QmWS1VAdMD353A6SDk9wNyvkT14kyCiZrNDYAad4w1tKqT/{lid}.json" and the client side is expected to replace the {id} as needed. 

Had an initial implementation that maps each token id to its own uri, but I believe if we use this approach, things will look fine on our app but might be broken on say Opensea if they follow the single uri standard above.

Reply:
Ignore above, after reading the standard again and seeing samples on OpenSea docs, they support both approaches. If the returned uri has the {id} string, it will handle by inserting the id on the client side. Otherwise, it would just use that is returned. Updated to track token uris for each token id now.